src/cmd_parser.cc

```diff
@@ -100,14 +118,68 @@ CmdParser::CmdParser(std::string _app_name, int argc, char **argv) {
     burn_in = getLastValueOrDefault(burn_in_, 0);
     stepsize = getLastValueOrDefault(stepsize_, 0.01);
     stepsize2 = getLastValueOrDefault(stepsize2_, 0.01);
+    if (stepsize == 0.01)
+      stepsize =
+          stepsize2;  // XXX hack to support two parameters to specify step size
     decay = getLastValueOrDefault(decay_, 0.95);
     reg_param = getLastValueOrDefault(reg_param_, 0.01);
-    regularization = getLastValueOrDefault(regularization_, std::string("l2"));
+    regularization =
+        getLastValueOrDefault(regularization_, std::string("l2")) == "l1"
+            ? REG_L1
+            : REG_L2;
 
+    should_use_snapshot = use_snapshot_->getValue() > 0;
     should_be_quiet = quiet_->getValue() > 0;
     should_sample_evidence = sample_evidence_->getValue() > 0;
     should_learn_non_evidence = learn_non_evidence_->getValue() > 0;
 
+  } else if (app_name == "text2bin") {
+    cmd_ = new TCLAP::CmdLine("DimmWitted text2bin", ' ', DimmWittedVersion);
+    TCLAP::UnlabeledValueArg<std::string> text2bin_mode_(
+        "mode", "what to convert", true, "",
+        "variable | domain | factor | weight", *cmd_);
+    TCLAP::UnlabeledValueArg<std::string> text2bin_input_(
+        "input", "path to an input file formatted in TSV, tab-separated values",
+        true, "/dev/stdin", "input_file_path", *cmd_);
+    TCLAP::UnlabeledValueArg<std::string> text2bin_output_(
+        "output", "path to an output file", true, "/dev/stdout",
+        "output_file_path", *cmd_);
+
+    //  factor-specific arguments
+    // TODO turn these into labeled args
+    TCLAP::UnlabeledValueArg<int> text2bin_factor_func_id_(
+        "func_id",
+        "factor function id (See: enum FACTOR_FUCNTION_TYPE in "
```

Comment:
Can we also please fix this typo `FACTOR_FUCNTION_TYPE` while we're doing the refactor? Every time I open `factor.h` it drives me nuts.